system-server/tests/persistence/test_persistent_generation.py

```diff
@@ -27,33 +25,24 @@ async def test_database_generation_on_init(tmpdir: Path) -> None:
     assert not expected.exists()
 
 
-async def test_create_tmpdir(monkeypatch: MonkeyPatch, tmpdir: Path) -> None:
-    # Mock out the settings
+async def test_persistent_directory_generation(tmpdir: Path) -> None:
+    # Test creating the directory when it exists
+    assert tmpdir == await create_persistent_directory(tmpdir)
+    assert tmpdir.exists()
 
-    class MockSettings:
-        def __init__(self, path: Path) -> None:
-            self.persistence_directory = path
+    subdir = tmpdir / "testing"
+    assert not subdir.exists()
+    assert subdir == await create_persistent_directory(subdir)
+    assert subdir.exists()
 
-    expected = tmpdir / "new_folder"
+    # Make sure opening the directory doesn't overwrite things
+    subfile = subdir / "test.txt"
+    assert not subfile.exists()
+    open(str(subfile), "w").write("Test string\n")
+    assert subdir == await create_persistent_directory(subdir)
+    assert open(subfile, "r").readline() == "Test string\n"
```

Comment:
To prevent file leakage:

```suggestion
    subfile.write_text("Test string\n")
    assert subdir == await create_persistent_directory(subdir)
    assert subfile.read_text() == "Test string\n"
```

Reply:
🚰